services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/SwitchOperationsBolt.java

```diff
@@ -0,0 +1,116 @@
+/* Copyright 2017 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.nbworker.bolts;
+
+import org.openkilda.messaging.info.event.SwitchInfoData;
+import org.openkilda.messaging.info.event.SwitchState;
+import org.openkilda.messaging.nbtopology.request.GetSwitchesRequest;
+import org.openkilda.messaging.nbtopology.request.SwitchesBaseRequest;
+import org.openkilda.pce.provider.Auth;
+import org.openkilda.wfm.AbstractBolt;
+
+import com.google.common.collect.Lists;
+import org.apache.storm.task.OutputCollector;
+import org.apache.storm.task.TopologyContext;
+import org.apache.storm.topology.OutputFieldsDeclarer;
+import org.apache.storm.tuple.Fields;
+import org.apache.storm.tuple.Tuple;
+import org.neo4j.driver.v1.AccessMode;
+import org.neo4j.driver.v1.Driver;
+import org.neo4j.driver.v1.Record;
+import org.neo4j.driver.v1.Session;
+import org.neo4j.driver.v1.StatementResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class SwitchOperationsBolt extends AbstractBolt {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SwitchOperationsBolt.class);
+
+    private Driver driver;
+    private final Auth neoAuth;
+
+    public SwitchOperationsBolt(Auth neoAuth) {
+        this.neoAuth = neoAuth;
+    }
+
+    @Override
+    public void prepare(Map stormConf, TopologyContext context, OutputCollector collector) {
+        this.driver = neoAuth.getDriver();
+        super.prepare(stormConf, context, collector);
+    }
+
+    protected void handleInput(Tuple input) {
```

Comment:
You should move common part that manage tuple unpacking and session management into base class (not AbstractBolt, create new intermediate one).

parseIntValue and similar methods should be moved into base class too.